### ISSUE TITLE
_minimize_writes is dangerous

### ISSUE THREAD

**Issue by mlschroe** [openSUSE/rpm-config-SUSE#30]
See https://github.com/rpm-software-management/rpm/pull/1355

Panu says the default should be off for now, as the code is too fragile.

**Comment by mlschroe** [comment]
See also https://bugzilla.redhat.com/show_bug.cgi?id=1872141

**Comment by lnussel** [comment]
The other view on the matter is that packages need to not do nasty things in scriptlets. I guess it's fair for rpm in general to be super safe and backwards compatible with all kinds of random 3rd party rpms. As distro we could require and adjust at least our own packages to be compliant with that mode though.

**Comment by pmatilai** [comment]
Yup, it's not that I believe in package's right to shoot themselves, the point is that an automatically enabled optimization must not alter transaction outcome, and that doesn't hold with the current _minimize_writes implementation in a number of scenarios.
